src/lib/es5.d.ts

```diff
@@ -943,14 +943,14 @@ interface JSON {
       * @param replacer A function that transforms the results.
       * @param space Adds indentation, white space, and line break characters to the return-value JSON text to make it easier to read.
       */
-    stringify(value: any, replacer?: (key: string, value: any) => any, space?: string | number): string;
+    stringify(value: any, replacer?: (key: string, value: any) => any, space?: string | number): string | undefined;

```

Comment:
I do not think you want everyone to have to handle `undefined` when calling `JSON.stringify`, it is just cases where hte input is `undefined` or possibly a function. for these, i would add overloads..e.g

```ts
stringify(value: number | string | boolean | null | object, replacer?: (key: string, value: any) => any, space?: string | number): string;
stringify(value: any, replacer?: (key: string, value: any) => any, space?: string | number): string | undefined;
```

Reply:
also pinging @DanielRosenwasser and @RyanCavanaugh for feedback

Reply:
Yeah, narrowing down when undefined my appear would be great. I'm afraid this overload wouldn't catch `function () {}`, because it is of type `object`. Calling `JSON.stringify(function () {})` would use the first provided overload, although the return value is `undefined`. [Example on TypeScript Playground](http://www.typescriptlang.org/play/#src=function%20giveMeAnObject(o%3A%20object)%20%7B%0D%0A%0D%0A%7D%0D%0A%0D%0AgiveMeAnObject(function%20()%20%7B%20%7D)%3B)
Is it even possible to distinguish functions from “normal” objects in JavaScript? This is an area where I have very little knowledge.
Perhaps we can add one more overload before the other two that uses the `Function` type?

Reply:
sure.